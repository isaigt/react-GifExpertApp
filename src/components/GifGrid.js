import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria }) => {
    const { data:images, loading } = useFetchGif( categoria );

    return (
        <>
            <h3 className="card animate__animated animate__fadeInDown">{ categoria }</h3>
            { loading && <h4 className="animate__animated animate__rotateInDownLeft">Cargando...</h4> }
            <div className="card-grid">
                    {
                        images.map( imagen => (
                            <GifGridItem key={ imagen.id } {...imagen} />
                        ))
                    }
            </div>
        </>
    );
}
