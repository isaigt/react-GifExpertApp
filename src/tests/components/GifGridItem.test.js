import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas del componente GifGridItem', () => {
    const title = "Prueba de titulo";
    const url = "https://localhost/prueba";
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> ); 

    test('se debe de renderizar correctamente el componente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('se debe de mostrar el titulo correcto', () => {
        const tituloWrapper = wrapper.find('p').text();
        expect( tituloWrapper ).toBe( title ); 
    });

    test('se debe mostrar la imagen con las propiedades correctas', () => {
        const imgWrapper = wrapper.find('img');
        expect( imgWrapper.prop('src') ).toBe( url ); 
        expect( imgWrapper.prop('alt') ).toBe( title ); 
    });

    test('el div debe tener la clase animate__fadeInUp', () => {
        const divWrapper = wrapper.find('div');
        expect( divWrapper.hasClass('animate__fadeInUp') ).toBe( true );
    });
});
