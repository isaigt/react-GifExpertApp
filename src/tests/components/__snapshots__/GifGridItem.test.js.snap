// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem se debe de renderizar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Prueba de titulo"
    src="https://localhost/prueba"
  />
  <p>
    Prueba de titulo
  </p>
</div>
`;
